sockeye/inference.py

```diff
@@ -1283,6 +1331,9 @@ def _beam_search(self,
         # scores_accumulated: chosen smallest scores in scores (ascending).
         scores_accumulated = mx.nd.zeros((self.batch_size * self.beam_size, 1), ctx=self.context)
 
+        # for constrained decoding
+        constraints = [[]] * self.batch_size * self.beam_size
```

Comment:
This might be safe in this context, but it's probably not good style. E.g.

```python
In [1]: aaa=[[]]*10

In [2]: aaa
Out[2]: [[], [], [], [], [], [], [], [], [], []]

In [3]: aaa[0].append(1)

In [4]: aaa
Out[4]: [[1], [1], [1], [1], [1], [1], [1], [1], [1], [1]]
```